### PR TITLE
build: use k8s 1.27 in devcontainer to match CI

### PR DESCRIPTION
**Pull request by agilgur5**
### Motivation

For dev/test parity
- https://github.com/argoproj/argo-workflows/pull/11267 updated CI to 1.27

### Modifications

- update devcontainer `k3d` to use `k3s` 1.27
  - found https://github.com/k3d-io/k3d/issues/92#issuecomment-1606249994 to specify the version
- update devcontainer `kubectl` to use 1.27

### Verification

Ran updated commands in my own devcontainer.

```
vscode ➜ ~/go/src/github.com/argoproj/argo-workflows (build-k8s-127) $ kubectl version
WARNING: This version information is deprecated and will be replaced with the output from kubectl version --short.  Use --output=yaml|json to get the full version.
Client Version: version.Info{Major:"1", Minor:"27", GitVersion:"v1.27.3", GitCommit:"25b4e43193bcda6c7328a6d147b1fb73a33f1598", GitTreeState:"clean", BuildDate:"2023-06-14T09:53:42Z", GoVersion:"go1.20.5", Compiler:"gc", Platform:"linux/amd64"}
Kustomize Version: v5.0.1
Server Version: version.Info{Major:"1", Minor:"27", GitVersion:"v1.27.3+k3s1", GitCommit:"fe9604cac119216e67f8bd5f14eb5608e3bcd3cf", GitTreeState:"clean", BuildDate:"2023-06-26T16:49:54Z", GoVersion:"go1.20.5", Compiler:"gc", Platform:"linux/amd64"}
```

### Future Work

Would be great to auto-check that these are in sync 😅

### CODE REVIEW

File: .devcontainer/pre-build.sh [hunk 1]
```diff
@@ -10,11 +10,11 @@ sudo bash -c 'echo "127.0.0.1 azurite" >> /etc/hosts'
 
 # install kubernetes
 wget -q -O - https://raw.githubusercontent.com/k3d-io/k3d/main/install.sh | bash
-k3d cluster get k3s-default || k3d cluster create --wait
+k3d cluster get k3s-default || k3d cluster create --image rancher/k3s:v1.27.3-k3s1 --wait
 k3d kubeconfig merge --kubeconfig-merge-default
 
 # install kubectl
-curl -LO https://dl.k8s.io/release/v1.26.0/bin/linux/amd64/kubectl
+curl -LO https://dl.k8s.io/release/v1.27.0/bin/linux/amd64/kubectl
```
Comment (terrytangyuan) [inline .devcontainer/pre-build.sh:17]: 1.27.3?

**Comment by terrytangyuan** [comment]
@agilgur5 Would you like to join ArgoProj as a member? If so, please follow instructions in https://github.com/argoproj/argoproj/blob/master/community/membership.md#member

**Comment by terrytangyuan** [comment]
Feel free to reach out to me on Slack if you have questions

**Comment by agilgur5** [comment]
Rebased on top of #11350 , which had a merge conflict with this

**Comment by agilgur5** [comment]
> @agilgur5 Would you like to join ArgoProj as a member? If so, please follow instructions in https://github.com/argoproj/argoproj/blob/master/community/membership.md#member

@terrytangyuan If you'll have me, I could contribute some more triage and reviews, but my contribution cycles tend to be quite sporadic. I don't often get dedicated time from work to contribute, so most of my OSS work is done in my free-time. On top of that, as I maintain a few libraries and contribute to several ecosystems, work tends to happen in bursts. This is one of those bursts 😅 

If you're alright with that limitation, it sounds like I'd still need a second sponsor

**Comment by terrytangyuan** [comment]
We can discuss on CNCF Slack. :-)

**Comment by agilgur5** [comment]
> We can discuss on CNCF Slack. :-)

For posterity, we briefly discussed over DMs and I filed https://github.com/argoproj/argoproj/issues/221 to request membership 🙂

**Pull request merged**